File: level_2/narrative2_sort_names_by_weight/langs/compiler.java
class Main
{
    public static void main(String[] args)
    {
        Scanner scanner = new Scanner(System.in);
    
        int n = scanner.nextInt();
        String[] names = new String[n];
        int[] weights = new int[n];
    
        for (int i = 0; i < n; i++)
            names[i] = scanner.next();
    
        for (int i = 0; i < n; i++)
            weights[i] = scanner.nextInt();

        Test test = new Test();
        String[] sortedNames = test.sort_names_by_weight(n, names, weights);
    
        for (int i = 0; i < n; i++)
            System.out.print(sortedNames[i] + " ");
    }
}

File: level_2/narrative2_sort_names_by_weight/langs/compiler.c
int main() {
    int n;
    scanf("%d", &n);

    char* names[n];
    int weights[n];

    for (int i = 0; i < n; i++) {
        names[i] = (char*)malloc(50 * sizeof(char));
        scanf("%s", names[i]);
    }

    for (int i = 0; i < n; i++)
        scanf("%d", &weights[i]);

    char** sortedNames = sortNamesByWeight(n, names, weights);

    for (int i = 0; i < n; i++) 
    {
        printf("%s ", sortedNames[i]);
        free(sortedNames[i]);
    }

    free(sortedNames);

    for (int i = 0; i < n; i++) 
    {
        free(names[i]);
    }

    return 0;
}
